apps/comments-ui/src/utils/helpers.ts

```diff
@@ -13,52 +13,53 @@ export function formatRelativeTime(dateString: string, t: TranslationFunction):
     const date = new Date(dateString);
     const now = new Date();
 
-    // Diff is in seconds
-    let diff = Math.round((now.getTime() - date.getTime()) / 1000);
-    if (diff < 5) {
+    // Handle invalid dates
+    if (isNaN(date.getTime())) {
         return t('Just now');
     }
 
-    if (diff < 60) {
-        return t('{{amount}} seconds ago', {amount: diff});
+    // Check if dates are on different calendar days by comparing UTC dates
+    const dateUTC = new Date(Date.UTC(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate()));
+    const nowUTC = new Date(Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate()));
+    const dayDiff = Math.floor((nowUTC.getTime() - dateUTC.getTime()) / (1000 * 60 * 60 * 24));
+
+    if (dayDiff === 1) {
+        return t('Yesterday');
     }
 
-    // Diff in minutes
-    diff = diff / 60;
-    if (diff < 60) {
-        if (Math.floor(diff) === 1) {
-            return t(`One min ago`);
+    if (dayDiff > 1) {
+        const day = date.getDate();
+        const month = date.toLocaleString('en-us', {month: 'short'});
+        
+        // If it's from a different year, include the year
+        if (date.getFullYear() !== now.getFullYear()) {
+            return `${day} ${month} ${date.getFullYear()}`;
         }
-        return t('{{amount}} mins ago', {amount: Math.floor(diff)});
+        // If it's from this year but not today/yesterday, show date without year
+        return `${day} ${month}`;
     }
 
-    // Diff in hours
-    diff = diff / 60;
-    if (diff < 24) {
-        if (Math.floor(diff) === 1) {
-            return t(`One hour ago`);
-        }
-        return t('{{amount}} hrs ago', {amount: Math.floor(diff)});
+    // If we're on the same calendar day, show relative time
+    let diff = Math.round((now.getTime() - date.getTime()) / 1000);
```

Comment:
Can we change this variable name to `secondDiff`, and likewise below instead of redefining `diff` create new variables like `minuteDiff` and `hourDiff`? That should make it a bit easier to understand the code without having to read backwards and keep track of the changing variable meaning